### DISCUSSION TITLE
Support for older Windows 7 and 8

### DISCUSSION THREAD

**Discussion by Kuraga13** [hecrj/iced#1872] [General]
Is it planned that iced can work with old Windows 7 and 8, will such support ever be implemented?
Is it possible to set something in the settings so as not to use anything in iced except for example native gui windows?

**Comment by Kuraga13** [comment]
I'll reformulate the question. Suppose I have a simple application with one window and a couple of buttons.
Can I use Iced to refer to something other than wgpu-opengl etc. but directly to windows native, using for example iced runtime and run the application on old win7-8 machines?
If it seems to you that this question is not relevant, unfortunately it is relevant.

**Reply by hecrj** [reply]
No, you cannot. And there is no planned support for interfacing with OS-native GUI toolkits.

**Comment by Kuraga13** [comment]
> No, you cannot. And there is no planned support for interfacing with OS-native GUI toolkits.

Maybe using your new tiny-ska renderer, or some 2d renderer can bypass the limitations of wgpu-directx, opengl-glow etc.?

**Reply by hecrj** [reply]
Which limitations? What is "windows native"?

**Reply by Kuraga13** [reply]
For example, I have to build two versions of PortableEXE - one with OpenGL, the other with wgpu.
OpenGL works on older machines, wgpu doesn't, somewhere wgpu build only works with anti-aliasing etc...

**Reply by hecrj** [reply]
Runtime renderer fallback recently landed in `master`. The new `tiny-skia` renderer is used as a fallback if `iced_wgpu` cannot be initialized.

**Reply by Kuraga13** [reply]
I got it, thanks! We will look for a solution for old machines, as in the field of electronics, hardware, etc. updating Windows is the last thing they will do to run a small application.

**Comment by Kuraga13** [comment]
>  What is "windows native"?

for example  https://github.com/gabdube/native-windows-gui

**Reply by hecrj** [reply]
As I said, there is no planned support for interfacing with OS-native GUI toolkits.
